### PR TITLE
Localstorage working

### PR DESCRIPTION
**Pull request by ekryski**
This gets all the tests passing (server side) and adds examples, etc. In theory because the node module api is the same as the browser it should work just fine there too.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -19,19 +19,27 @@ class LocalStorage extends Service {
   }
   
   create(... args) {
-    return super.create(... args).then(this.flush);
+    return super.create(... args).then(data => {
+      return this.flush.call(this, data);
```
Comment (daffl) [inline src/index.js]: Shorthand: `return super.create(... args).then(data => this.flush(data));`

File: test/index.test.js [hunk 2]
```diff
@@ -1,13 +1,52 @@
...
+    }, done);
+  });
+
+  afterEach(done => {
+    const doneNow = () => done();
+    people.remove(_ids.Doug).then(doneNow, doneNow);
+  });
 
-describe('feathers-localstorage', () => {
   it('is CommonJS compatible', () => {
```
Comment (daffl) [inline test/index.test.js]: We might want to set up `after()` to make sure that something has been written to the storage.

File: src/index.js (hunk 1 above)
Comment (ekryski) [inline src/index.js]: I'm pretty sure I tried that and it didn't work. I'll double check.

File: test/index.test.js (hunk 2 above)
Comment (ekryski) [inline test/index.test.js]: Not sure I follow. Shouldn't the tests already be testing that stuff was written to storage?

File: src/index.js [hunk 3]
```diff
@@ -1,37 +1,60 @@
...
-    );
+    this.storage = options.storage || window.localStorage;
+    this.throttle = options.throttle || 200;
+    this.store = null;
+  }
+  
+  ready() {
+    if(!this.store) {
+      return Promise.resolve(this.storage.getItem(this.storageKey))
+        .then(str =>JSON.parse(str || '{}'))
```
Comment (ekryski) [inline src/index.js]: I think we might have to have a try catch here in case someone pre-loads their localstorage data with invalid JSON.

**Comment by ekryski** [comment]
https://github.com/feathersjs/feathers-errors/pull/26 will need to be landed and `feathers-memory` will need to be updated to the next major `feathers-errors` version before this can be merged.

**Comment by daffl** [comment]
Good thing is that `feathers-memory` will only be a patch release. Plus we can change the Lodash dependencies so that it doesn't pull in the whole thing.

**Comment by ekryski** [comment]
Still need to get a new version of `feathers-memory` in without all of lodash before we merge this. https://github.com/feathersjs/feathers-memory/pull/18

**Pull request merged**